js/src/util/backdrop.js

```diff
@@ -98,10 +98,16 @@ class Backdrop {
       return
     }
 
-    this._config.rootElement.appendChild(this._getElement())
+    let { rootElement, clickCallback } = this._config
+
+    if (typeof rootElement === 'function') {
+      rootElement = rootElement()
+    }
+
+    rootElement.appendChild(this._getElement())
```

Comment:
How about, instead of function, allow `null` and do just `(rootElement || document.body).appendChild(...` It should be less code.

Reply:
Beside fixing the error, I was thinking about to make it a little bit more dynamic. So maybe one could use it like 
```js
new Backdrop({
  rootElement: () => document.querySelector('[bs-data-backdrop-root]')
})
```
and it would evaluate the selector right before the backdrop is opened. Currently Backdrop isn't using any events, so this would also allow to trigger something right before it shows.

But actually there is currently no need for this. It was rather some idea (that would fix the issue as well).

I also missed that there is already another PR opened, that rather follows your approach, so I will close this.